src/System.Buffers.Primitives/System/Buffers/Buffer.cs

```diff
@@ -69,24 +70,19 @@ public Buffer(T[] array, int start, int length)
             if ((uint)start > (uint)array.Length || (uint)length > (uint)(array.Length - start))
```

Comment:
Is casting to uint necessary? 
It looks like: 
`if ((uint)length > array.Length - (uint)start)`  is the same, because if (start > array.Length)  ->  array.Length - (uint)start less than zero and so less than any uint.